ioengine/IOEngine.py

```diff
@@ -0,0 +1,53 @@
+"""The IOEngine class."""
+import inspect
+from types import ModuleType
+import sys
+from utils import dotdict
+
+
+class IOEngine:
+    """The IOEngine class"""
+
+    def __init__(self):
+        self.services = []
+        self.engine_api = dotdict(showMessage=print, register=self.register_service)
```

Comment:
Please use lowercase snakecase for attribute names, ie change `showMessage` to `show_message`.